lib/rules/no-whitespace-within-word.js

```diff
@@ -105,7 +105,29 @@ function splitTextByEntity(input) {
 module.exports = class NoWhitespaceWithinWord extends Rule {
   visitor() {
     return {
+      Program: {
+        enter() {
+          this.attrNodeCount = 0;
+        },
+        exit() {
+          delete this.attrNodeCount;
+        },
+      },
+
+      AttrNode: {
+        enter() {
+          this.attrNodeCount++;
+        },
+        exit() {
+          this.attrNodeCount--;
+        },
+      },
+
       TextNode(node) {
+        if (this.attrNodeCount > 0) {
+          return;
+        }
+
```

Comment:
```suggestion
      TextNode(node, path) {
        let parents = [...path.parents()];
        if (parents.find((parent) => parent.node.type === 'AttrNode')) {
          return;
        }
```

Reply:
That looks great!  I had no idea that there was the second "path" argument.